### DISCUSSION TITLE
Suddenly, 2 of my 3 throttles have no throttle function.

### DISCUSSION THREAD

**Discussion by Budman1758** [DCC-EX/EX-T3-WiFi#104] [Q&A] [answered]
I have no idea why but suddenly 2 of my 3 units have no throttle up or down function. Forward\reverse and controlling track power works fine. Also controlling functions on the loco's works fine too. Just no increase or decrease in engine speed when turning the rotary encoder. Both units started doing this at the same time.

Only thing different was I updated the command station to the latest version. All of these things were out of use for a few months so I first did the update to the command station. Was using the first throttle I built for a week or so then got the other 2 out to copy loco configs from #1 to the others. This is when I became aware of this issue.

I also tried reflashing the 2 throttles but that seems to have made no difference. As far as I can tell the only function not working as it should is the rotary encoder up\down function. Can't really imagine two encoders failing the exact same way at the exact same time....

Any idea how to go about troubleshooting this?  Thanks.

All of 3 of these are built exactly the same using the components listed in this project.

**Comment by alex-code** [comment]
Just a few Q's to see if we can narrow down the issue.

1. Is it just 2 of the 3 you made not working?
2. Are they all connected to the CS at the same time?
3. Which board did you use for your CS?

**Comment by Budman1758** [comment]
#1 Yes

 #2  Either way. Connected individually. all works but speed up or down. If all 3 are connected and connected to the same loco they all update and show the same info and all will control the loco EXCEPT for the 2 will still not control speed. They show the current speed that the #1 throttle is set to and update when it changes.

#3 ELEGOO MEGA R3 Board ATmega 2560
   [https://www.amazon.com/ELEGOO-ATmega2560-ATMEGA16U2-Projects-Compliant/dp/B01H4ZLZLQ/?th=1](url)

**Reply by alex-code** [reply]
You could try adding `Serial.println("CW");` between lines 481 and 482 in `main.cpp` then use the Upload and monitor option.

If encoder rotation is detected it should print to the serial monitor.

**Comment by Budman1758** [comment]
Apparently its not detecting the rotation. Throttle that works fine prints to the console. The others don't.

Guess I will see if I can get a couple new rotary encoders....

**Reply by alex-code** [reply]
Do you think any of the wires could have come loose?

If you've got a multimeter you can do a basic check.
If you put the multimeter in continuity mode and put one probe on GND and the other to either pin A or B then turn the encoder slowly the multimeter should beep.

Depending on how you assembled them you could try swapping the encoder on the working one to see if it works in one of the other 2.

**Reply by Budman1758** [reply]
Well.... Put the working encoder into one of the non-working throttles.  Throttle does not work.  Put non-working encoder into the throttle that does work and the encoder works fine. 

I cannot find any problems with the wiring in the 2 that don't work.

This is getting weirder by the minute.

**Reply by alex-code** [reply]
Are you able to check the continuity of the wires from the encoder to the firebeetle?

It's a lot more involved but you could try swapping the firebeetle's too. Seems unlikely two would develop pin issues but this could help narrow down the issue.

**Comment by Budman1758** [comment]
Sorry for the delay.  Life got in the way.

Swapped the fire beetle from the non working (#2) to the working throttle (#1) and it works as it should. Problem NOT the fire beetle.

Working (#1) with fire beetle (#2) works fine.

Non-working throttle (#2) with brand new rotary encoder does not work. Problem unchanged.

Wiring is good on the #2 throttle. Checked it point to point and also with multimeter and I can see that the encoder is sending pulses to the fire beetle. All wiring checks out OK to the best of ability to test it.

When all 3 throttles ore on and connected to the same locomotive they all update when you change anything like lights or direction. They all will control those things also. Only throttle #1 will change loco speed.

I will try redoing the wiring on the 2 non-working ones.  It all checks out with my meter, but I'll give it a whirl.....

**Reply by Budman1758** [reply]
Re wired #2 and no change.

How do I do a complete wipe of the fire beetle? Maybe something is funked up and does not get written over when I upload the code.

BTW... When all 3 throttles are connected to the same loco the throttle response on the #1 throttle that works is considerably impacted. Very slow response. Turn the others off and response is very quick.  Just an FYI for what its worth. Not a likely scenario during real operation.

**Comment by Budman1758** [comment, marked as answer]
Well.... I'm a friggin idiot.  Guss I never tested these 2 back when they were built.  Had one of the wires on the wrong pin on the fire beetle from the encoder. Green wire was right next to the white instead of one pin over.  Sorry for the BS here. I go sit in the corner with my dunce cap on now.....

**Reply by alex-code** [reply]
Haha, one of my next suggestions was going to be a pic of your wiring.
Glad you got it sorted 👍

RE the slowness with multiple throttles there's an issue with the WiFi shields that I can't work around.
When broadcasting to all connected clients there's a delay.

I'm hoping this'll be better with an ESP32 based CS.
